### PR TITLE
Configuration rewrite (WIP)

### PR DESCRIPTION
**Pull request by copygirl**
Just a work in progress for reviewing purposes.
See issue #16. Will probably include #27 later here, too.

### CODE REVIEW

File: src/Alpacka.Lib/Pack/Config/EntryDefaults.cs [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+using System.Collections.Generic;
+using System.Reflection;
+using YamlDotNet.Core;
+using YamlDotNet.Serialization;
+
+namespace Alpacka.Lib.Pack.Config
+{
+    public class EntryDefaults : ICollection<EntryDefaults.Group>
+    {
+        public EntryDefaults()
```
Comment (NikkyAI) [inline src/Alpacka.Lib/Pack/Config/EntryDefaults.cs]: so why is that stuff hardcoded and there is this same info in the default.yaml ? is this supposed to be all implied ?

**Comment by NikkyAI** [comment]
tested just now.. it seems to have issues reading the defaults block
tested with this 18d9e6da9316a9d42ad5f6ef7d74ec55bf04ce96
the error is
```
Unhandled Exception: YamlDotNet.Core.YamlException: (Line: 15, Col: 3, Idx: 405) - (Line: 15, Col: 3, Idx: 405): Expected 'SequenceStart', got 'MappingStart' (at Line: 15, Col: 3, Idx: 405).
   at YamlDotNet.Core.ParserExtensions.Expect[T](IParser parser)
   at YamlDotNet.Serialization.NodeDeserializers.CollectionNodeDeserializer.DeserializeHelper(Type tItem, IParser parser, Func`3 nestedObjectDeserializer, IList result, Boolean canUpdate)
   at YamlDotNet.Serialization.NodeDeserializers.CollectionNodeDeserializer.YamlDotNet.Serialization.INodeDeserializer.Deserialize(IParser parser, Type expectedType, Func`3 nestedObjectDeserializer, Object&
value)
   at YamlDotNet.Serialization.ValueDeserializers.NodeValueDeserializer.DeserializeValue(IParser parser, Type expectedType, SerializerState state, IValueDeserializer nestedObjectDeserializer)
   at YamlDotNet.Serialization.ValueDeserializers.AliasValueDeserializer.DeserializeValue(IParser parser, Type expectedType, SerializerState state, IValueDeserializer nestedObjectDeserializer)
   at YamlDotNet.Serialization.NodeDeserializers.ObjectNodeDeserializer.YamlDotNet.Serialization.INodeDeserializer.Deserialize(IParser parser, Type expectedType, Func`3 nestedObjectDeserializer, Object& valu
e)
   at Alpacka.Lib.Utility.ValidatingNodeDeserializer.Deserialize(IParser reader, Type expectedType, Func`3 nestedObjectDeserializer, Object& value) in /home/nikky/dev/alpacka/src/Alpacka.Lib/Utility/Validati
ngNodeDeserializer.cs:line 18
   at YamlDotNet.Serialization.ValueDeserializers.NodeValueDeserializer.DeserializeValue(IParser parser, Type expectedType, SerializerState state, IValueDeserializer nestedObjectDeserializer)
   at YamlDotNet.Serialization.ValueDeserializers.AliasValueDeserializer.DeserializeValue(IParser parser, Type expectedType, SerializerState state, IValueDeserializer nestedObjectDeserializer)
   at YamlDotNet.Serialization.Deserializer.Deserialize(IParser parser, Type type)
   at YamlDotNet.Serialization.Deserializer.Deserialize[T](TextReader input)
   at Alpacka.Lib.Pack.Config.ModpackConfig.LoadYAML(String path) in /home/nikky/dev/alpacka/src/Alpacka.Lib/Pack/Config/ModpackConfig.cs:line 34
   at Alpacka.CLI.Commands.CommandBuild.<>c.<.ctor>b__0_0() in /home/nikky/dev/alpacka/src/Alpacka.CLI/Commands/CommandBuild.cs:line 26
   at Microsoft.Extensions.CommandLineUtils.CommandLineApplication.Execute(String[] args)
   at Alpacka.CLI.Program.Run(String[] args) in /home/nikky/dev/alpacka/src/Alpacka.CLI/Program.cs:line 35
   at Alpacka.CLI.Program.Main(String[] args) in /home/nikky/dev/alpacka/src/Alpacka.CLI/Program.cs:line 10
```

File: src/Alpacka.Lib/Pack/Config/EntryDefaults.cs (hunk 1 above)
Comment (copygirl) [inline src/Alpacka.Lib/Pack/Config/EntryDefaults.cs]: Yes. These are the default `defaults` (which can be overwritten, I hope).

**Comment by NikkyAI** [comment]
this creates at the moment duplicate mod entries due to curse dependencies that do not get removed

**Pull request merged**